macros/WebExtBrowserCompat.ejs

```diff
@@ -1,13 +1,60 @@
 <%
 
-/*
+/**
 WebExtBrowserCompat.ejs
 -----------------------
 Use this macro to insert the appropriate WebExtension compat table for the
 current page.
 
-It transforms the URL of the current page into a path through the compat
-data, and uses this path to retrieve the appropriate bit of data.
+It takes one optional argument.
+
+* If the argument is given, the macro calls tableFromArgument(), which uses
+the argument to get the right chunk of JSON data for the table. Currently
+this function only supports tables for manifest.json keys.
+
+* If the argument is omitted, the macro calls tableFromSlug(), which uses
+the slug of the current page to get the right chunk of JSON data for the
+table. The no-argument form is only allowed on pages under
+"Add-ons/WebExtensions/API/".
+*/
+
+/**
+The tableFromArgument() function takes a single argument, which must be
+a string that traces a path through the browser compat data.
+The string may be either of:
+
+    * "data.webextensions.manifest". In this case the function generates an
+    agregate table for all manifest keys.
+
+    * "data.webextensions.manifest.<KEY-NAME>", where <KEY-NAME> is the name
+    of a manifest key, like "applications" or "content_scripts". In this case
+    the function generates a table for the given manifest key.
+*/
+function tableFromArgument(path) {
+  var path = $0; // "data.webextensions.manifest.applications"
+  var pathComponents = path.split(".");
+  // first, validate the argument
+  // there must be at least "data", "webextensions", "manifest"
+  if (pathComponents.length < 3 ||
+    !((pathComponents[0] === "data") && (pathComponents[1] === "webextensions") && (pathComponents[2] === "manifest"))) {
+    throw "If an argument is given it must point to WebExtensions manifest data.";
+  }
+  const manifestDataUrl = "https://raw.githubusercontent.com/mdn/browser-compat-data/master/webextensions/manifest-keys.json";
+  // if the argument does not specify a manifest key, generate the aggregate table
+  if (pathComponents.length === 3) {
+    const manifestBaseUrl = "Add-ons/WebExtensions/manifest.json/";
```

Comment:
On the aggregate table I get URLs like http://localhost:8000/en-US/Add-ons/WebExtensions/manifest.json//applications, so this could be without the trailing slash I think.